tests/unit/jobserver/test_icons.py

```diff
@@ -1,30 +1,38 @@
-import itertools
 import re
 from pathlib import Path
 
+import pytest
 
-def test_all_icons(subtests):
-    """Ensure icon tag names defined in templates are valid tags"""
-    templates = [p for p in Path("templates").glob("**/*.html")]
 
-    # remove templates with no icon tag so we can simplify the regex matching
-    # code below
-    icon_templates = [p for p in templates if "{% icon_" in p.read_text()]
-
-    # look for the start of an icon templatetag and capture the full name
-    icon_pat = re.compile(r"{% (icon_[^\s]+)")
-
-    # get a set of icon names in the Django templates
-    icons = set(
-        itertools.chain.from_iterable(
-            icon_pat.search(t.read_text()).groups() for t in icon_templates
+# Logically, this feels like it could be a pytest.fixture.
+# However, it's not currently possible to use fixtures with parametrize:
+# https://github.com/pytest-dev/pytest/issues/349
+def find_icon_tag_names_used():
+    """Find the name of template tags starting with `icon_` used in our templates."""
+    icon_pattern = re.compile(r"{% (icon_[^\s]+)")
+    template_paths = list(Path("templates").rglob("*.html"))
+    # Get a sorted list of non-duplicated icon names in the Django templates.
```

Comment:
```suggestion
    # Get a sorted list of all of the icon names used in the Django templatetags.
```

"non-duplicated" could be interpreted as "occurring exactly once" which is not what we mean here.

Reply:
Thanks; I think this makes it clearer. Have included this and the other suggestion in the single commit.